git_machete/client.py

```diff
@@ -577,7 +578,8 @@ def slide_out(self,
         self.__git.checkout(new_upstream)
         for new_downstream in new_downstreams:
             self.__git.checkout(new_downstream)
-            if opt_merge:
+            use_merge = opt_merge or (new_downstream in self.annotations and self.annotations[new_downstream].qualifiers.merge)
```

Comment:
Actually... as for now, the qualifiers don't apply outside of `traverse`. There's an issue for that which should be rather quick to do: https://github.com/VirtusLab/git-machete/issues/769. I've even assigned it to the current milestone v3.25.0... there's just always something else that pops up 😅 currently I'm adding GitLab integration alongside GitHub

Reply:
Although... this qualifier has a different semantics anyway, so maybe leave as indeed